feat: implement LocaleCubit for language management

abdo-essam opened the pull request:
Integrates `LocaleCubit` to handle app-wide language changes and persistence using `SharedPreferences`.

- Introduces `LocaleCubit` and `LocaleState` for managing the application's locale.
- Refactors `ChangeLanguageBottomSheet` to use `LocaleCubit` for state management, removing local state handling.
- Updates `main.dart` to provide `LocaleCubit` to the widget tree and respond to locale changes.
- Removes the previous `LocalizationService` in favor of the new Cubit implementation.

## Description
Brief description of changes made in this PR

## What Changed
-

## Screenshots/Videos (if applicable)
<!-- Add screenshots or videos to help explain your changes -->

## Checklist
- [ ] My code follows the project's style guidelines
- [ ] Changes have been tested manually and verified.
- [ ] PR includes at most one single feature.

## Additional Notes

abdo-essam committed:
feat: implement LocaleCubit for language management

Integrates `LocaleCubit` to handle app-wide language changes and persistence using `SharedPreferences`.

- Introduces `LocaleCubit` and `LocaleState` for managing the application's locale.
- Refactors `ChangeLanguageBottomSheet` to use `LocaleCubit` for state management, removing local state handling.
- Updates `main.dart` to provide `LocaleCubit` to the widget tree and respond to locale changes.
- Removes the previous `LocalizationService` in favor of the new Cubit implementation.

Commit:
Branch was auto-updated.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

abdo-essam committed:
Merge remote-tracking branch 'origin/develop' into feature/enable-the-logic-of-switch-language-in-acoount-screen

# Conflicts:
#	lib/presentation/screens/account/account_screen.dart

abdo-essam committed:
refactor: improve account screen state handling and UI

Refactors the `AccountScreen` to better handle different states (`Loading`, `Error`, `Loaded`).

Specific changes include:
- Replaces the monolithic `_buildBody` method with distinct builder methods for loading, error, and loaded states (`_buildLoadingInfo`, `_buildErrorInfo`, `_buildLoadedInfo`).
- Introduces a dedicated UI for the error state, including an error message and a "Try Again" button.
- Implements a shimmer loading effect for the user information section.
- Simplifies the main body widget by using `if/else if` conditions to render the appropriate UI based on the current `AccountState`.

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

mohamedebrahem13 commented on lib/core/localization/l10n/app_localizations.dart:
```diff
@@ -104,12 +104,372 @@ abstract class AppLocalizations {
   /// **'Guest'**
   String get guest;
 
+  /// No description provided for @home.
```
remove unnecessary comment

abdo-essam committed:
Delete generated localization

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.
